src/3pc/util.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2021 MIT Digital Currency Initiative,
+//                    Federal Reserve Bank of Boston
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "util.hpp"
+
+#include <unordered_map>
+
+namespace cbdc::threepc {
+    auto split(const std::string& s, const std::string& delim)
+        -> std::vector<std::string> {
+        size_t pos_start{};
+        size_t pos_end{};
+        std::vector<std::string> ret;
+
+        while((pos_end = s.find(delim, pos_start)) != std::string::npos) {
+            auto token = s.substr(pos_start, pos_end - pos_start);
+            pos_start = pos_end + delim.size();
+            ret.emplace_back(token);
+        }
+
+        ret.emplace_back(s.substr(pos_start));
+        return ret;
+    }
+
+    auto parse_args(int argc, char** argv)
```

Comment:
Agreed. Also, keeping in mind that corresponding changes to test controller are likely required.

Reply:
I mentioned this as an issue without offering a clear action to close review item; sorry about that!

In short, I mentioned this because I think the addition of this architecture exacerbates the existing DIY configuration system, which motivates our trying to solve the problem. I think a new issue targeting a unifying configuration system for the three architectures we have, and with space for more. (Probably referencing #173 and/or #81.) At some point, before merge, that issue should be created (or establishing an alternative path toward handling this inconsistency; for example, an official sunset, or stable release plan---these are not the only options, of course).

Hopefully that clarifies the purpose of this task, as well as a clear completion target.

Reply:
#218